test/genReqId.test.js

```diff
@@ -72,3 +72,74 @@ test('Custom genReqId function gets raw request as argument', t => {
     })
   })
 })
+
+test('Should accept option to set genReqId with setGenReqId option', t => {
```

Comment:
Can you add a tests where a `plugin` alters the `genReqId` of the parent?

Reply:
Do we want the `plugin` to alter the `genReqId` of parent? Added test for opposite [here](https://github.com/fastify/fastify/pull/5259/commits/11a42ea0a87c13dba1dacdc52f535d08564e769b#diff-c59d02cf0c0d124c85b50aa4a031acfb2d5e176b9a8be295b90125929d64bd1bR335) but can change this if we want `plugin` to update `parent.genReqId`

Reply:
It should actually not alter the parent?! 
@metcoder95 please clarify ;)

Reply:
haha, yeah to alter the parent. The test you point out tests when creating a new encapsulated context, which it **should not alter the parent**.
But, for plugins that does not create a new encapsulated context, they usually **seek to alter the parent context**

Reply:
@metcoder95 add in test to set on root scope using fastify-plugin.
This is failing however and not working as expected should I follow how `setErrorHandler` and `setChildLoggerFactory` work to fix this?